src/core/stats.cpp

```diff
@@ -22,9 +22,108 @@
 namespace pyston {
 
 #if !DISABLE_STATS
-std::vector<long>* Stats::counts;
+extern "C" const char* getStatTimerNameById(int id) {
+    return Stats::getStatName(id).c_str();
+}
+
+extern "C" int getStatTimerId() {
+    return StatTimer::getStack()->getId();
+}
+
+extern "C" const char* getStatTimerName() {
+    return getStatTimerNameById(getStatTimerId());
+}
+
+__thread StatTimer* StatTimer::stack;
+StatTimer::StatTimer(int statid, bool push) {
+    uint64_t at_time = getCPUTicks();
+    _duration = 0;
+    _start_time = 0;
+    _statid = statid;
+
+    if (!push) {
+        _prev = NULL;
+        return;
+    }
+
+    _prev = stack;
+    stack = this;
+    if (_prev) {
+        _prev->pause(at_time);
+    }
+    resume(at_time);
+}
+
+StatTimer::StatTimer(int statid, uint64_t at_time) {
+    _duration = 0;
+    _start_time = 0;
+    _statid = statid;
+    _prev = stack;
+    stack = this;
+    if (_prev) {
+        _prev->pause(at_time);
+    }
+    resume(at_time);
+}
+
+StatTimer::~StatTimer() {
+    assert(stack == this);
+
+    uint64_t at_time;
+    if (!isPaused()) {
+        at_time = getCPUTicks();
+        pause(at_time);
+    } else {
+        // fprintf (stderr, "WARNING: timer was paused.\n");
+        at_time = _last_pause_time;
```

Comment:
ah, turns out we do need the gymnastics in jit.cpp.  without them the ticks don't match up and we get a warning.  sweet.